bitmapcontainer.go

```diff
@@ -1234,3 +1231,15 @@ func (bc *bitmapContainer) addOffset(x uint16) (container, container) {
 
 	return low, high
 }
+
+// validate checks that the container size is non-negative
+func (bc *bitmapContainer) validate() error {
+	if bc.cardinality < arrayDefaultMaxSize {
+		return fmt.Errorf("bitmap container size was less than: %d", arrayDefaultMaxSize)
+	}
+	if bc.cardinality > maxCapacity {
+		return fmt.Errorf("bitmap container size was greater than: %d", maxCapacity)
+	}
```

Comment:
Because the `bitmapContainer` is implemented as a slice, we would also need to check that the slice length (multiplied by 64) doesn't exceed the `maxCapacity`.

Reply:
We would also need to check consistency I think. That is, `bc.cardinality` must match the real cardinality of the inner container.

Reply:
@bearrito These added checks seem reasonable. They are both easy checks:

- `bc.cardinality = int(popcntSlice(bc.bitmap))`,
- `len(bc.bitmap)*64 == maxCapacity`.

Or something similar.